tests/api/ocsp.c

```diff
@@ -0,0 +1,539 @@
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+
+    conf.resp = (unsigned char*)resp_bad_noauth;
+    conf.respSz = sizeof(resp_bad_noauth);
+    conf.ca0 = root_ca_cert_pem;
+    conf.ca0Sz = sizeof(root_ca_cert_pem);
+    conf.ca1 = ca_cert_pem;
+    conf.ca1Sz = sizeof(ca_cert_pem);
+    conf.targetCert = server_cert_pem;
+    conf.targetCertSz = sizeof(server_cert_pem);
+    ret = test_ocsp_response_with_cm(&conf);
+#ifndef WOLFSSL_NO_OCSP_ISSUER_CHECK
+    ExpectIntNE(ret, WOLFSSL_SUCCESS);
+#else
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+#endif
+    return EXPECT_SUCCESS();
+}
+#else  /* HAVE_OCSP */
+int test_ocsp_response_parsing(void) { return TEST_SKIPPED; }
+#endif /* HAVE_OCSP */
+
+#if defined(HAVE_OCSP) && (defined(OPENSSL_ALL) || defined(OPENSSL_EXTRA))
+static int test_ocsp_create_x509store(WOLFSSL_X509_STORE** store,
+    unsigned char* ca, int caSz)
+{
+    EXPECT_DECLS;
+    WOLFSSL_X509* cert = NULL;
+    int ret;
+
+    *store = wolfSSL_X509_STORE_new();
+    ExpectPtrNE(*store, NULL);
+    cert = wolfSSL_X509_d2i(&cert, ca, caSz);
+    ExpectPtrNE(cert, NULL);
+    ret = wolfSSL_X509_STORE_add_cert(*store, cert);
+    wolfSSL_X509_free(cert);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    return EXPECT_RESULT();
+}
+
+static int test_create_stack_of_x509(WOLF_STACK_OF(WOLFSSL_X509) * *certs,
+    unsigned char* der, int derSz)
+{
+    EXPECT_DECLS;
+    WOLFSSL_X509* cert = NULL;
+    int ret;
+
+    *certs = wolfSSL_sk_X509_new_null();
+    ExpectPtrNE(*certs, NULL);
+    cert = wolfSSL_X509_d2i(&cert, der, derSz);
+    ExpectPtrNE(cert, NULL);
+    ret = wolfSSL_sk_X509_push(*certs, cert);
+    ExpectIntEQ(ret, 1);
+    return EXPECT_RESULT();
+}
+
+int test_ocsp_basic_verify(void)
+{
+    EXPECT_DECLS;
+    WOLF_STACK_OF(WOLFSSL_X509) * certs;
+    OcspResponse* response = NULL;
+    WOLFSSL_X509_STORE* store;
+    const unsigned char* ptr;
+    DecodedCert cert;
+    int ret;
+
+    wc_InitDecodedCert(&cert, ocsp_responder_cert_pem,
+        sizeof(ocsp_responder_cert_pem), NULL);
+    ret = wc_ParseCert(&cert, CERT_TYPE, 0, NULL);
+    ExpectIntEQ(ret, 0);
+
+    /* just decoding */
+    ptr = (const unsigned char*)resp;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp));
+    ExpectPtrNE(response, NULL);
+    ExpectIntEQ(response->responseStatus, 0);
+    ExpectIntEQ(response->responderIdType, OCSP_RESPONDER_ID_NAME);
+    ExpectBufEQ(response->responderId.nameHash, cert.subjectHash,
+        OCSP_DIGEST_SIZE);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* responder Id by key hash */
+    ptr = (const unsigned char*)resp_rid_bykey;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_rid_bykey));
+    ExpectPtrNE(response, NULL);
+    ExpectIntEQ(response->responseStatus, 0);
+    ExpectIntEQ(response->responderIdType, OCSP_RESPONDER_ID_KEY);
+    ExpectBufEQ(response->responderId.keyHash, cert.subjectKeyHash,
+        OCSP_DIGEST_SIZE);
+    wc_FreeDecodedCert(&cert);
```

Comment:
Free it at the very end. After this `cert` can't be access but its not a pointer so its hard to catch invalid access.